client/ayon_core/pipeline/load/plugins.py

```diff
@@ -251,15 +252,50 @@ class ProductLoaderPlugin(LoaderPlugin):
     """
 
 
+class PrePostLoaderHookPlugin:
+    """Plugin that should be run before or post specific Loader in 'loaders'
+
+    Should be used as non-invasive method to enrich core loading process.
+    Any studio might want to modify loaded data before or after
+    they are loaded without need to override existing core plugins.
+    """
+    @classmethod
+    def is_compatible(cls, Loader: LoaderPlugin) -> bool:
+        pass
+
+    def pre_load(
+        self,
+        context: dict,
+        name: str | None = None,
+        namespace: str | None = None,
+        options: dict | None = None,
+    ):
+        pass
+
+    def post_load(
+        self,
+        context: dict,
+        name: str | None = None,
+        namespace: str | None = None,
+        options: dict | None = None,
+        container: dict | None = None,  # (ayon:container-3.0)
+    ):
+        pass
+
+    def switch(self, container, context):
+        pass
+
+
 def discover_loader_plugins(project_name=None):
     from ayon_core.lib import Logger
     from ayon_core.pipeline import get_current_project_name
 
     log = Logger.get_logger("LoaderDiscover")
-    plugins = discover(LoaderPlugin)
     if not project_name:
         project_name = get_current_project_name()
     project_settings = get_project_settings(project_name)
+    plugins = discover(LoaderPlugin)
+    hooks = discover(PrePostLoaderHookPlugin)
```

Comment:
We might just create objects here and not?
```suggestion
    hooks = [cls() for cls in discover(PrePostLoaderHookPlugin)]
```